app/models/ability.rb

```diff
@@ -4,21 +4,22 @@ class Ability
   include CanCan::Ability
 
   def initialize(user)
-    # Guest users have no privileged access
-    return unless user.present?
+    role = user&.role
 
-    case user.role
-    when :admin
+    if role == :admin
       can :manage, :all
-    when :events
+    elsif role == :events && user.active?
```

Comment:
Is this only to lock down what events staff can see? Where did this requirement come from?
Because now once someone is able to see the RDF for the submissions related to their assigned event, they can also see everything else. Wouldn't it make sense to lock the `can show, Story` down with a block similar to the `:edit` one, plus any `published?` stories?

Reply:
An events user should only have privileged access when they are assigned to one or more event. The intention is that after staff from an event have completed any work on contributions from their events, they will be unassigned from the events, making them inactive, and preventing any further changes by them. This is a business requirement.